File: codegen/tarantool/fixtures/server/handlers/users_post_handler.lua
-- THIS FILE IS SAFE TO EDIT. It will not be overwritten when rerunning go-raml.

local schema = require("handlers.schemas.schema")

function users_post_handler(request)
    -- handler for POST /users
    -- request body should match schema.User
    -- response body for 200 should match schema.User

    local resp = {
    }
    return resp
end

return users_post_handler

File: codegen/python/gentest/goramldir/server/handlers/users_postHandler.py
# THIS FILE IS SAFE TO EDIT. It will not be overwritten when rerunning go-raml.

from flask import jsonify, request

import json as JSON
import jsonschema
from jsonschema import Draft4Validator

import os

dir_path = os.path.dirname(os.path.realpath(__file__))
User_schema = JSON.load(open(dir_path + '/schema/User_schema.json'))
User_schema_resolver = jsonschema.RefResolver('file://' + dir_path + '/schema/', User_schema)
User_schema_validator = Draft4Validator(User_schema, resolver=User_schema_resolver)


def users_postHandler():

    inputs = request.get_json()

    try:
        User_schema_validator.validate(inputs)
    except jsonschema.ValidationError as e:
        return jsonify(errors="bad request body"), 400

    return jsonify(inputs)
